Add logo image url to search results

MatthewPiatetsky opened the pull request:
@edx/ecommerce 

We want to expose the authoring organization's logo in the new list view

mikedikan commented on course_discovery/apps/course_metadata/search_indexes.py:
```diff
@@ -200,6 +206,10 @@ class ProgramIndex(BaseIndex, indexes.Indexable, OrganizationsMixin):
 
+    def prepare_logo_image_url(self, obj):
+        org = obj.authoring_organizations.first()
+        return org.logo_image_url if org else ''
```
curious why empty string instead of null?

clintonb commented on course_discovery/apps/course_metadata/search_indexes.py:
```diff
@@ -65,10 +65,15 @@ class BaseCourseIndex(OrganizationsMixin, BaseIndex):
     partner = indexes.CharField(model_attr='partner__short_code', null=True, faceted=True)
 
+    def prepare_logo_image_url(self, obj):
+        org = obj.authoring_organizations.first()
```
Given the our data supports multiple authoring organizations, we should index them and return them from the API. Leave it up to the client to decide which to use. We should not artificially limit ourselves, especially given that we know we eventually want to support multiple authoring organizations.

MatthewPiatetsky replied:
I got a different impression from Jasper, but updated

MatthewPiatetsky commented on course_discovery/apps/course_metadata/search_indexes.py:
```diff
@@ -200,6 +206,10 @@ class ProgramIndex(BaseIndex, indexes.Indexable, OrganizationsMixin):
+    def prepare_logo_image_url(self, obj):
+        org = obj.authoring_organizations.first()
+        return org.logo_image_url if org else None
+
```
fixed

MatthewPiatetsky commented:
@clintonb @davec-edx have I addressed your comments?

davec-edx commented:
👍

MatthewPiatetsky commented:
@clintonb @davec-edx have I addressed your comments? fyi I realized that we already had access to logo_image_url for programs, so I only added for course runs

clintonb commented on course_discovery/apps/course_metadata/search_indexes.py:
```diff
@@ -65,10 +65,15 @@ class BaseCourseIndex(OrganizationsMixin, BaseIndex):
 
+    def prepare_logo_image_urls(self, obj):
+        orgs = obj.authoring_organizations.all()
+        return [org.logo_image_url for org in orgs if org]
```
`return list(obj.authoring_organizations.values_list('logo_image_url', flat=True))`

MatthewPiatetsky replied:
last time I did this I noticed that values_list ended up generating additional SQL queries, this seems simpler rather than doing that and prefetching

clintonb commented on course_discovery/apps/api/tests/test_serializers.py:
```diff
@@ -1038,6 +1039,7 @@ def test_data(self):
             'published': course_run.status == CourseRunStatus.Published,
             'partner': course_run.course.partner.short_code,
             'program_types': course_run.program_types,
+            'logo_image_urls': [org.logo_image_url for org in orgs if org],
```
The `if` statement is not necessary.
